lib/ansible/modules/network/ios/ios_logging.py

```diff
@@ -148,7 +148,7 @@ def map_obj_to_commands(updates, module):
 
         if state == 'absent' and w in have:
             if dest == 'host':
-                commands.append('no logging host {}'.format(name))
+                commands.append('no logging host {} transport udp port 514'.format(name))
```

Comment:
why hardcode port with 514 ?

Reply:
As far as I can tell, ios_logging currently does the following to configure a log host
```
logging host 1.2.3.4
```
In the complete running-config this is then configured as follows
```
#conf t
Enter configuration commands, one per line.  End with CNTL/Z.
(config)#logging host 1.2.3.4
(config)#end
#show run all | sec logging host
logging host 1.2.3.4 transport udp port 514
#
```

Therefore, the proper way to reliably remove the command configured by the module would be to explicitely state the commands as shown above. If this is not done, other commands could be removed as well.

Of course, a proper fix would be to have the port and transport configurable via the module, then this does not have to be hardcoded. Until then, the more correct way to remove the configuration done by ios_logging is to hardcode the port and transport.

Reply:
Makes sense, if the same server is configured on non-default port by non-Ansible method.

However, for a complete solution please enhance  the module args to accept transport and port (default them to: udp and 514 respectively). 

Deletion of specific `transport:port` combo for `host` should happen only if those args are provided, else it should delete all configs for that `host`

`host` is a key when transport and port are not provided, whereas `transport` becomes a key when transport is provided. (port will be unique on that node so that does not apply)

Please add more commits with above changes along with related unit/integration tests.